docs/usage/self-hosted-configuration.md

```diff
@@ -125,6 +124,12 @@ If using negations, all repositories except those who match the regex are added
 }
 ```
 
+**Negation logic**
+
+- All "positive match" filters (not starting with `!`) are ORed together so that only one needs to match
+- All "negative match" filters (those starting with `!`) are ANDed together so that all need to match
+- The aggregate results of the negative and positive matches are ANDed together
```

Comment:
I've accidentally bumped into it and as an author of #21347 which would be "covered" by that MR, I think that it would be good to document some usage of that approach. E.g:

```json
{
  "autodiscoverFilter": [
    "whole-group1/*",
    "!/whole-group1/legacy-project-to-skip/"
  ]
}
```

Reply:
Btw, I've update the sample filter to cover more complex case: in `whole-group1` there are Git repositories with pure environment configurations (nothing to handle by Renovate), with the convention: `ops/test1-config`, `ops/prod-config`, however, they could nested in the `whole-group1`. In the end, I wonder, if my filter(s) are properly crafted or there is a need to quote for example `/` in the middle?